server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CCSUsageTelemetryIT.java

```diff
@@ -0,0 +1,466 @@
+        assertThat(perCluster.size(), equalTo(3));
+        for (String clusterAlias : remoteClusterAlias()) {
+            var clusterTelemetry = perCluster.get(clusterAlias);
+            assertThat(clusterTelemetry.getCount(), equalTo(1L));
+            assertThat(clusterTelemetry.getSkippedCount(), equalTo(0L));
+            assertThat(clusterTelemetry.getTook().count(), equalTo(1L));
+        }
+
+        // another search
+        assertResponse(cluster(LOCAL_CLUSTER).client(nodeName).search(searchRequest), Assert::assertNotNull);
+        telemetry = getTelemetrySnapshot(nodeName);
+        assertThat(telemetry.getTotalCount(), equalTo(2L));
+        assertThat(telemetry.getSuccessCount(), equalTo(2L));
+        assertThat(telemetry.getFailureReasons().size(), equalTo(0));
+        assertThat(telemetry.getTook().count(), equalTo(2L));
+        assertThat(telemetry.getTookMrtTrue().count(), equalTo(minimizeRoundtrips ? 2L : 0L));
+        assertThat(telemetry.getTookMrtFalse().count(), equalTo(minimizeRoundtrips ? 0L : 2L));
+        assertThat(telemetry.getRemotesPerSearchAvg(), equalTo(2.0));
+        assertThat(telemetry.getRemotesPerSearchMax(), equalTo(2L));
+        assertThat(telemetry.getSkippedRemotes(), equalTo(0L));
+        assertThat(telemetry.getClientCounts().size(), equalTo(1));
+        perCluster = telemetry.getByRemoteCluster();
+        assertThat(perCluster.size(), equalTo(3));
+        for (String clusterAlias : remoteClusterAlias()) {
+            var clusterTelemetry = perCluster.get(clusterAlias);
+            assertThat(clusterTelemetry.getCount(), equalTo(2L));
+            assertThat(clusterTelemetry.getSkippedCount(), equalTo(0L));
+            assertThat(clusterTelemetry.getTook().count(), equalTo(2L));
+        }
+    }
+
+    /**
+     * Search on a specific remote
+     */
+    public void testOneRemoteSearch() throws ExecutionException, InterruptedException {
+        Map<String, Object> testClusterInfo = setupClusters();
+        String localIndex = (String) testClusterInfo.get("local.index");
+        String remoteIndex = (String) testClusterInfo.get("remote.index");
+
+        // Make request to cluster a
+        SearchRequest searchRequest = makeSearchRequest(localIndex, REMOTE1 + ":" + remoteIndex);
+        String nodeName = cluster(LOCAL_CLUSTER).getRandomNodeName();
+        assertResponse(cluster(LOCAL_CLUSTER).client(nodeName).search(searchRequest), Assert::assertNotNull);
+        CCSTelemetrySnapshot telemetry = getTelemetrySnapshot(nodeName);
+        var perCluster = telemetry.getByRemoteCluster();
+        assertThat(perCluster.size(), equalTo(2));
+        assertThat(perCluster.get(REMOTE1).getCount(), equalTo(1L));
+        assertThat(perCluster.get(REMOTE1).getTook().count(), equalTo(1L));
+        assertThat(perCluster.get(REMOTE2), equalTo(null));
+
+        // Make request to cluster b
+        searchRequest = makeSearchRequest(localIndex, REMOTE2 + ":" + remoteIndex);
```

Comment:
Unless I missed it, I don't think there are any tests in here that do not search the local cluster. It is quite common for clusters to do CCS where they query only remote clusters, so some of your test cases, should only search remotes to make sure we don't have some bug that doesn't handle telemetry if the local cluster was not searched.

Reply:
OK, I didn't think not searching the local would make much difference, but sure, I can add such test too, it makes sense.

Reply:
Added